flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgram.java

```diff
@@ -220,12 +220,47 @@ public ClassLoader getUserCodeClassLoader() {
 	 * Returns all provided libraries needed to run the program.
 	 */
 	public List<URL> getJobJarAndDependencies() {
-		List<URL> libs = new ArrayList<URL>(this.extractedTempLibraries.size() + 1);
+		return getJobJarAndDependencies(jarFile, extractedTempLibraries, isPython);
+	}
+
+	/**
+	 * Returns all provided libraries needed to run the program.
+	 */
+	public static List<URL> getJobJarAndDependencies(File jarFile, @Nullable String entryPointClassName) throws ProgramInvocationException {
```

Comment:
In this class you add this method only, so the rest of the changes can be reverted, right. Essentially the `public List<URL> getJobJarAndDependencies()` can stay as is currently on the master and you can remove the `List<URL> getJobJarAndDependencies(URL jarFileUrl, List<File> extractedTempLibraries, boolean isPython)`.

Reply:
fixed